src/app/Http/Controllers/API/PollController.php

```diff
@@ -89,7 +110,13 @@ public function destroy(Poll $poll)
      */
     public function publish(Request $request, Poll $poll)
     {
+        $req_user = $poll->users()->find($request->user());
+        if (!($req_user && $req_user->role == 'presenter')) {
+            return response()->json('unauthorized', Response::HTTP_UNAUTHORIZED);
+        }
```

Comment:
Have you tested your code?

Reply:
Cannot test due to PresentationUserController lacking logic, so cannot make user subscribe to presentation.